src/kernel/kmain.zig

```diff
@@ -45,13 +48,17 @@ export fn kmain(mb_info: *multiboot.multiboot_info_t, mb_magic: u32) void {
         var buffer = mem_profile.vaddr_end[0..mem_profile.fixed_alloc_size];
         var fixed_allocator = std.heap.FixedBufferAllocator.init(buffer);
 
+        panic_root.init(&mem_profile, &fixed_allocator.allocator) catch |e| {
+            panic_root.panic(@errorReturnTrace(), "Failed to initialise panic: {}", .{e});
+        };
+
         pmm.init(&mem_profile, &fixed_allocator.allocator);
+        kernel_vmm = vmm.init(&mem_profile, mb_info, &fixed_allocator.allocator, arch.VmmPayload, arch.VMM_MAPPER, arch.KERNEL_VMM_PAYLOAD) catch |e| panic_root.panic(@errorReturnTrace(), "Failed to initialise kernel VMM: {}", .{e});
```

Comment:
Yes but the VMM should really be agnostic to what the low-level details are. Doing it this way means we can easily pass process-specific stuff later on.

Reply:
I just realised that of course this is referring to the top-level init function rather than the one that initialises an individual VMM. I can make the top-level init function get them itself.

Reply:
But it will be getting the top level arch anyway bc vmm.zig isn't in the arch folder.

Reply:
Oh no I mean the top-level init function in vmm.zig, so `vmm.init` rather then `vmm.VirtualMemoryManager.init`.